### PR TITLE
Fix CF on X86_INS_SHL

### PR DESCRIPTION
**Pull request by xtrm0**

### CODE REVIEW

File: libr/anal/p/anal_x86_cs.c [hunk 1]
```diff
@@ -777,6 +777,31 @@ static void anop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 		}
 		break;
 	case X86_INS_SHL:
+	{
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: ths seems like spaguetti with the other PR, can you generalize the patch to reuse this switch case?

**Comment by xtrm0** [comment]
Ok, then let me make a new PR with the change, to make mergine easier (and also because october fest)

**Comment by xtrm0** [comment]
I think I will just move the switch case, because SHL and SAL are exactly the same instruction in x86 afaik

**Comment by xtrm0** [comment]
Also shouldn't there be some unit tests for the ESIL + ESIL emulator? Like for instance setting values on registers on qemu, then running a few steps, and then comparing some values with the values of the same operations on the ESIL?

**Comment by radare** [comment]
can we close this ?

**Pull request closed without merging**